### PR TITLE
UHM-6614 Move Mexico Diagnosis concepts from MDS to Initializer for CES

### PR DESCRIPTION
**Pull request by brandones**
https://pihemr.atlassian.net/browse/UHM-6614

Removes diagnoses from Mexico MDS and adds them to an Initializer concepts CSV.

All mappings are listed as SAME-AS mappings.

No problems with updating the server or starting from scratch, aside from the discovery of https://github.com/PIH/openmrs-config-ces/pull/96 .

### CODE REVIEW

**Comment by mseaton** [comment]
I'd like us to consider limiting this file to just diagnoses, and putting the sets in a separate file, ideally in a separate domain.  I added a "conceptsets" domain at the end of last year when I was trying to enhance initializer to spike on supporting an OCL -> Iniz CSV workflow.  I never got around to doing the same for mappings, hence the recent discussions around this and the ticket @brandones created.

Personally, I think it'll be easier to manage open-ended 1:many relationships in separate files/domains than trying to cram them into a single CSV.  The "Members" column in this CSV is impossibly unwieldy and impossible to read.  That should be moved to the conceptsets domain.

For mappings, I'd like to avoid the ";"-delimited list of source:code mappings in favor of something more structured.  I do think there is value in having the PIH-mapping or CIEL mapping (and for a csv of diagnoses, the ICD-10 mapping, etc) right in the concepts csv since you often want to refer to these easily.  In those cases, I'd like to see us expand the mapping header options in a similar way to how "Index term" and "Synonym" headers work - we'd support a parseable header like "Mapping Type|Source|Unique Suffix", so you would have columns like "SAME-AS|CIEL", "SAME-AS|PIH|CODE", "SAME-AS|PIH|NAME", etc.  This way, you could keep your core CIEL and PIH mappings in the concept csv, and move any other mappings that might only have a few associations across different sources for groups of concepts to dedicated conceptreferenceterm csv files.

**Review by mseaton — CHANGES_REQUESTED** [review]
See comments.

**Comment by mseaton** [comment]
One other comment - there is no need to have the Numeric and Complex header columns in this CSV, so we should remove them if not applicable.

**Comment by mseaton** [comment]
> It's a small difference, but I think for single-set files like this, managing the set in the same file as the contents is a bit easier.

Ultimately, if some one wants to do this, and they are the main ones authoring and supporting the file and it's deployment and troubleshooting, the option is there.  But in terms of setting us up with an initial set of files that feels best practice, managing our sets as massive semi-colon delimited strings of names is not what I would want to recommend.  If we need to support all of these in one file, I'd rather we have an additional column in the concepts csv that allows a concept to declare that it should be added as a member of a particular set.  Something like a column in the format:  Member-of|COVID-19 diagnosis set, where concepts you want to add to this set have a value of "true".

Regarding some of the other comments, there is value beyond what one can do in Excel we need to think about.  How these files appear in github, how easily changes and diffs can be assessed in them, etc. are also considerations.

I don't really know what you are referring to regarding the "rarely used columns".  If columns aren't used in a particular csv at all, they should not be added.

**Comment by brandones** [comment]
> I don't really know what you are referring to regarding the "rarely used columns". If columns aren't used in a particular csv at all, they should not be added.

Sorry, I was referring to mappings like SNOMED-CT.

**Comment by Joseph-Developer** [comment]
> Es una pequeña diferencia, pero creo que para archivos de un solo conjunto como este, administrar el conjunto en el mismo archivo que los contenidos es un poco más fácil. El flujo de trabajo por si un montón de conceptos cambian es más o menos el mismo:
> 
> * Si el conjunto está en el archivo, simplemente cree la fórmula `=JOIN(";", K1:K200)`y luego copie y pegue los resultados como texto sin formato en la celda "miembros del conjunto".
> * Si el conjunto está en un archivo de conjuntos de conceptos, elimine el contenido existente de la columna "Miembro" de ese archivo y reemplácelo con K1:K200, luego extienda las otras columnas según sea necesario.
> 
> Cualquiera de los dos es "ilegible" en el sentido de que no puede mirarlo e identificar de inmediato lo que falta o debe eliminarse. En cualquier caso, debe presionar Ctrl-F "buscar".
> 
> Entonces, la principal diferencia es si tiene uno o dos archivos para jugar cuando necesita hacer un cambio. No creo que el valor estético de los conjuntos de conceptos justifique su uso, en este caso particular.
> 
> Pero esta es una pequeña diferencia, no es realmente un gran problema para mí.
> 
> Con respecto a las asignaciones que se usan con poca frecuencia: si no desea ver una columna, haga clic con el botón derecho en ella y haga clic en "ocultar columna". O toma su borde y hazlo muy delgado. Dividir las asignaciones que no le interesan a nadie en su propio archivo aumentaría sustancialmente la carga de mantenimiento asociada con esas asignaciones, que nuevamente, a nadie le importa.
> 
> CC@Joseph-Developer

yes, of course, I certainly think that this would be faster to do when wanting to implement new concepts, like the csv that is used for drugs would be something similar, right?

**Comment by brandones** [comment]
Yes @Joseph-Developer, it would be like the CSV for drugs.

**Comment by brandones** [comment]
@mseaton Reformatting and refactoring the CSV files should be very easy. Once there is support for splitting up mappings into separate columns, or adding a concept to a set using a column, you should be able to refactor the CSV with a formula and a few clicks in Excel. So I think the germane issue is whether or not we're okay with making all mappings SAME-AS for now. Once other mapping types are supported in Initializer, the appropriate support can be added to the concept CSV exporter, a new CSV can be re-exported, and the mappings in the original CSV can be updated using a VLOOKUP comparison.

I am a big fan of the idea, by the way, of letting concepts declare what sets they belong to. I think that would be a very elegant solution to the problem.

**Comment by brandones** [comment]
I've re-exported the CSV with all the mappings of different types. Aside from that all the changes we were talking about are just formatting changes that can be done in Excel.

File: configuration/concepts/diagnoses.csv [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
...gus;Desgarro de menisco;Choque eléctrico;Ruptura del tendón de Aquilles;Malformación congénita de la espina dorsal;Dislocación del codo;Pie cavo;Talo equinovaro;Infección postraumática de una herida;Poliartropatía inflamatoria;Escoliosis congénita por malformación osea congénita;Artritis juvenil;Artritis gonocócica;Lordosis;Fractura patellar;Malformación congénita de la mano;Dislocación congénita unilateral de cadera;Hidrocefalia;cálculo renal;Hemoptisis;neuralgia;retinoblastoma;Linfoma maligno (clínico);himen imperforado;Infarto cerebral;Anafilaxia;Sepsis;Hepatitis viral aguda tipo B;Sinusitis crónica;Inoxicación alimentaria;Tuberculosis pulmonar;síndrome HELLP;VIH en el embarazo;desprendimiento placentario;Amenaza de parto prematuro;prolapso de cordón - nacido;Recién nacido pretermino;Recién nacido macrosómico;Recién nacido postérmino;Recién nacido vivo;Generalmente sano;Enfermedad infecciosa;Blefaritis;Neuropatía;Amputación;Pterigión;Orzuelo;Diarrea viral;Faringitis;Pólipo nasal;Hernia femoral;Estreñimiento;Diarrea aguda;Presbicia;Insuficiencia venosa periférica;Faringitis estreptocócica;Hernia con obstrucción;Faringoamigdalitis",PIH:11722;PIH:Mexico primary care diagnosis set,,
```
Comment (mogoodrich) [inline configuration/concepts/diagnoses.csv:581]: Sorry, belated comment here... I think @brandones and @mseaton  were going back and forth on this, and I agree with @mseaton  that having a giant cell of semicolon delimited concept names to maintain set membership does seem like a real pain to maintain (vs a separate file).  However, I think I'd also be okay with starting like this and seeing if our fears are warrantied.

**Review by mseaton — APPROVED** [review]
This is fine w/ me for step 1, to get things moved over to iniz, assuming the before and after are now the exact same now that we have the mapping type enhancement.

I do think we should explore formatting changes to this in further phases, and experiment with whether there are significant improvements or regressions in management if we were to move each individual mapping to it's own column, and if we were to introduce the Member Of:Set X column feature.  That can be done as a future refactoring if and when desired.

**Pull request merged**